tests/data/permutations/all_roman_numbers.csv

```diff
@@ -0,0 +1,3999 @@
+3824,MMMDCCCXXIV
+3825,MMMDCCCXXV
+3826,MMMDCCCXXVI
+3827,MMMDCCCXXVII
+3828,MMMDCCCXXVIII
+3829,MMMDCCCXXIX
+3830,MMMDCCCXXX
+3831,MMMDCCCXXXI
+3832,MMMDCCCXXXII
+3833,MMMDCCCXXXIII
+3834,MMMDCCCXXXIV
+3835,MMMDCCCXXXV
+3836,MMMDCCCXXXVI
+3837,MMMDCCCXXXVII
+3838,MMMDCCCXXXVIII
+3839,MMMDCCCXXXIX
+3840,MMMDCCCXL
+3841,MMMDCCCXLI
+3842,MMMDCCCXLII
+3843,MMMDCCCXLIII
+3844,MMMDCCCXLIV
+3845,MMMDCCCXLV
+3846,MMMDCCCXLVI
+3847,MMMDCCCXLVII
+3848,MMMDCCCXLVIII
+3849,MMMDCCCXLIX
+3850,MMMDCCCL
+3851,MMMDCCCLI
+3852,MMMDCCCLII
+3853,MMMDCCCLIII
+3854,MMMDCCCLIV
+3855,MMMDCCCLV
+3856,MMMDCCCLVI
+3857,MMMDCCCLVII
+3858,MMMDCCCLVIII
+3859,MMMDCCCLIX
+3860,MMMDCCCLX
+3861,MMMDCCCLXI
+3862,MMMDCCCLXII
+3863,MMMDCCCLXIII
+3864,MMMDCCCLXIV
+3865,MMMDCCCLXV
+3866,MMMDCCCLXVI
+3867,MMMDCCCLXVII
+3868,MMMDCCCLXVIII
+3869,MMMDCCCLXIX
+3870,MMMDCCCLXX
+3871,MMMDCCCLXXI
+3872,MMMDCCCLXXII
+3873,MMMDCCCLXXIII
+3874,MMMDCCCLXXIV
+3875,MMMDCCCLXXV
+3876,MMMDCCCLXXVI
+3877,MMMDCCCLXXVII
+3878,MMMDCCCLXXVIII
+3879,MMMDCCCLXXIX
+3880,MMMDCCCLXXX
+3881,MMMDCCCLXXXI
+3882,MMMDCCCLXXXII
+3883,MMMDCCCLXXXIII
+3884,MMMDCCCLXXXIV
+3885,MMMDCCCLXXXV
+3886,MMMDCCCLXXXVI
+3887,MMMDCCCLXXXVII
+3888,MMMDCCCLXXXVIII
+3889,MMMDCCCLXXXIX
+3890,MMMDCCCXC
+3891,MMMDCCCXCI
+3892,MMMDCCCXCII
+3893,MMMDCCCXCIII
+3894,MMMDCCCXCIV
+3895,MMMDCCCXCV
+3896,MMMDCCCXCVI
+3897,MMMDCCCXCVII
+3898,MMMDCCCXCVIII
+3899,MMMDCCCXCIX
+3900,MMMCM
+3901,MMMCMI
+3902,MMMCMII
+3903,MMMCMIII
+3904,MMMCMIV
+3905,MMMCMV
+3906,MMMCMVI
+3907,MMMCMVII
+3908,MMMCMVIII
+3909,MMMCMIX
+3910,MMMCMX
+3911,MMMCMXI
+3912,MMMCMXII
+3913,MMMCMXIII
+3914,MMMCMXIV
+3915,MMMCMXV
+3916,MMMCMXVI
+3917,MMMCMXVII
+3918,MMMCMXVIII
+3919,MMMCMXIX
+3920,MMMCMXX
+3921,MMMCMXXI
+3922,MMMCMXXII
+3923,MMMCMXXIII
+3924,MMMCMXXIV
+3925,MMMCMXXV
+3926,MMMCMXXVI
+3927,MMMCMXXVII
+3928,MMMCMXXVIII
+3929,MMMCMXXIX
+3930,MMMCMXXX
+3931,MMMCMXXXI
+3932,MMMCMXXXII
+3933,MMMCMXXXIII
+3934,MMMCMXXXIV
+3935,MMMCMXXXV
+3936,MMMCMXXXVI
+3937,MMMCMXXXVII
+3938,MMMCMXXXVIII
+3939,MMMCMXXXIX
+3940,MMMCMXL
+3941,MMMCMXLI
+3942,MMMCMXLII
+3943,MMMCMXLIII
+3944,MMMCMXLIV
+3945,MMMCMXLV
+3946,MMMCMXLVI
+3947,MMMCMXLVII
+3948,MMMCMXLVIII
+3949,MMMCMXLIX
+3950,MMMCML
+3951,MMMCMLI
+3952,MMMCMLII
+3953,MMMCMLIII
+3954,MMMCMLIV
+3955,MMMCMLV
+3956,MMMCMLVI
+3957,MMMCMLVII
+3958,MMMCMLVIII
+3959,MMMCMLIX
+3960,MMMCMLX
+3961,MMMCMLXI
+3962,MMMCMLXII
+3963,MMMCMLXIII
+3964,MMMCMLXIV
+3965,MMMCMLXV
+3966,MMMCMLXVI
+3967,MMMCMLXVII
+3968,MMMCMLXVIII
+3969,MMMCMLXIX
+3970,MMMCMLXX
+3971,MMMCMLXXI
+3972,MMMCMLXXII
+3973,MMMCMLXXIII
+3974,MMMCMLXXIV
+3975,MMMCMLXXV
+3976,MMMCMLXXVI
+3977,MMMCMLXXVII
+3978,MMMCMLXXVIII
+3979,MMMCMLXXIX
+3980,MMMCMLXXX
+3981,MMMCMLXXXI
+3982,MMMCMLXXXII
+3983,MMMCMLXXXIII
+3984,MMMCMLXXXIV
+3985,MMMCMLXXXV
+3986,MMMCMLXXXVI
+3987,MMMCMLXXXVII
+3988,MMMCMLXXXVIII
+3989,MMMCMLXXXIX
+3990,MMMCMXC
+3991,MMMCMXCI
+3992,MMMCMXCII
+3993,MMMCMXCIII
+3994,MMMCMXCIV
+3995,MMMCMXCV
+3996,MMMCMXCVI
+3997,MMMCMXCVII
+3998,MMMCMXCVIII
```

Comment:
I [read](https://blog.prepscholar.com/roman-numerals-converter#:~:text=*Though%20normally%20considered%20incorrect%20usage,numerals%20is%20MMMCMXCIX%2C%20or%203%2C999.) that `MMMCMXCIX` would be a valid `3999`.